src/rust/ide/view/graph-editor/src/component/visualization/container/visualization_chooser.rs

```diff
@@ -0,0 +1,315 @@
+//! Definition of the icon + menu displayed to choose the visualisation for a
+//! node.
+
+use crate::prelude::*;
+
+use crate::component::visualization;
+
+use enso_frp as frp;
+use enso_frp;
+use ensogl::application::Application;
+use ensogl::data::color;
+use ensogl::display::shape::*;
+use ensogl::display::traits::*;
+use ensogl::display;
+use ensogl::gui::component;
+use ensogl_gui_list_view as list_view;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Invisible dummy color to catch hover events.
+const HOVER_COLOR : color::Rgba = color::Rgba::new(1.0,0.0,0.0,0.000_001);
+/// The width of the visualisation selection menu.
+const MENU_WIDTH  : f32         = 180.0;
+
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+/// Icon that indicates the drop down menu.
+pub mod icon {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+            let width            = Var::<Pixels>::from("input_size.x");
+            let height           = Var::<Pixels>::from("input_size.y");
+            let triangle         = Triangle(width,height);
+            let triangle_down    = triangle.rotate(Var::<f32>::from(std::f32::consts::PI));
+            let color_path       = ensogl_theme::vars::graph_editor::visualization::action_bar::icon::color;
+            let icon_color       = style.get_color(color_path);
+            let triangle_colored = triangle_down.fill(color::Rgba::from(icon_color));
+
+            triangle_colored.into()
+        }
+    }
+}
+
+/// Invisible rectangular area around the icon
+pub mod chooser_hover_area {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        () {
+            let width  : Var<Pixels> = "input_size.x".into();
+            let height : Var<Pixels> = "input_size.y".into();
+            let background           = Rect((&width,&height));
+            let background           = background.fill(HOVER_COLOR);
+            background.into()
+        }
+    }
+}
+
+
+
+// ===============================
+// === Visualisation Path List ===
+// ===============================
+
+/// The `VisualisationPathList` allows us to show a `Vec<visualization::Path>` in the
+/// `list_view::ListView` by implementing `list_view::entry::ModelProvider`.
+#[derive(Clone,Debug,Default)]
+struct VisualisationPathList {
+    content: Vec<visualization::Path>
+}
```

Comment:
Wrapping vector of displayable things seems to me like a very common pattern, thence I would consider making a batch-implementation of `list_view::entry::ModelProvider` for every `Vec<T:Into<Model>>` and `From<T:Display>` for `Model` - then you will not need such implementations.

Additionally, `ListView` structure could have helper method `set_entries(&self, entries:impl entry::ModelProvider)` which just cast entries and emit frp endpoint.